src/js/jsx/sections/style/Radius.jsx

```diff
@@ -44,68 +44,137 @@ define(function (require, exports, module) {
 
-        shouldComponentUpdate: function (nextProps) {
+        shouldComponentUpdate: function (nextProps, nextState) {
+            return this.state !== nextState;
+        },
+
+        componentWillReceiveProps: function (nextProps) {
             var getRelevantProps = function (props) {
-                var layers = props.document.layers.selected;
+                    var layers = props.document.layers.selected.filter(function (layer) {
+                        return layer.radii;
+                    });
 
-                return collection.pluckAll(layers, ["id", "bounds", "radii"]);
-            };
+                    return collection.pluckAll(layers, ["id", "bounds", "radii"]);
+                },
+                relevantProps = getRelevantProps(this.props),
+                nextRelevantProps = getRelevantProps(nextProps);
+
+            if (!Immutable.is(relevantProps, nextRelevantProps)) {
+                this._calculateState(nextProps);
+            }
+        },
 
-            return !Immutable.is(getRelevantProps(this.props), getRelevantProps(nextProps));
+        componentWillMount: function () {
+            this._calculateState(this.props);
+        },
+
+        /**
+         * Given the props object, calculates and sets the state
+         * This is refactored out of componentWillReceiveProps, because 
+         * it doesn't run on initial render, causing a null state
+         *
+         * @private
+         * @param {object} props
+         */
+        _calculateState: function (props) {
+            var layers = props.document.layers.selected.filter(function (layer) {
+                    return layer.radii;
+                }),
+                scalars = layers.map(function (layer) {
+                    return layer.radii.scalar || 0;
+                }),
+                maxRadius = layers
+                    .toSeq()
+                    .filter(function (layer) {
+                        return !!layer.bounds;
+                    })
+                    .reduce(function (sides, layer) {
+                        return sides.concat(Immutable.List.of(layer.bounds.width / 2, layer.bounds.height / 2));
+                    }, Immutable.List())
+                    .min();
+
+            this.setState({
+                layers: layers,
+                scalars: scalars,
+                maxRadius: maxRadius,
+                maxRadiusInput: Math.floor(maxRadius)
+            });
+        },
+
+        _handleScrubStart: function () {
+            var currentRadius = collection.uniformValue(this.state.scalars);
+
+            // This initial call helps set the history state right before coalescing
+            if (currentRadius !== null) {
+                this.setState({
+                    scrubRadius: currentRadius
+                });
+
+                this.startCoalescing();
+            }
+        },
+
+        _handleRadiusScrub: function (deltaX) {
+            if (this.state.scrubRadius === null) {
```

Comment:
this does double duty: if scrubRadius is null, it could be that there is non-uniform value... or simply that setState (which is async) has completed.  Nothing wrong here, just pointing it out (wrt to my prev comment)